src/Tribe/Template.php

```diff
@@ -584,6 +619,90 @@ public function get_template_file( $name ) {
 		return false;
 	}
 
+	/**
+	 * Runs the entry point hooks and filters
+	 *
+	 * @param string  $entry_point_name The name of the entry point.
+	 * @param boolean $echo             If we should also print the entry point content.
+	 *
+	 * @return null|string
+	 */
+	public function do_entry_point( $entry_point_name, $echo = true ) {
+		$hook_name = $this->get_template_current_hook_name();
+
+		/**
+		 * Filter if the entry points are enabled.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean $is_enabled       Is entry_point enabled.
+		 * @param string  $hook_name        For which template include this entry point belongs.
+		 * @param string  $entry_point_name Which entry point specifically we are triggering.
+		 * @param self    $template         Current instance of the template class doing this entry point.
+		 */
+		$is_entry_point_enabled = apply_filters( 'tribe_template_entry_point_is_enabled', true, $hook_name, $entry_point_name, $this );
+
+		if ( ! $is_entry_point_enabled ) {
+			return null;
+		}
+
+		ob_start();
+
+		/**
+		 * Generic entry point action for the current template.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $hook_name        For which template include this entry point belongs.
+		 * @param string $entry_point_name Which entry point specifically we are triggering.
+		 * @param self   $template         Current instance of the template class doing this entry point.
+		 */
+		do_action( "tribe_template_entry_point:$hook_name", $hook_name, $entry_point_name, $this );
```

Comment:
I would check if there's any listener for this action before running it using `has_action`
The same I would do for the rest of these actions.
Filters, and actions, are costly operations with a relevant performance overhead price even if nothing is listening on them; furthermore, the `all` filter will always fire.
The `has_filter` and `has_action` functions are cheap and can save a lot of wasted processing.